### PR TITLE
Feature/authenticate

### PR DESCRIPTION
**Pull request by trabnm1313**
- Added authenticate to certain path.
- Fix path due to authenticator middleware function recognize all of routers as same path.
WIP waiting for another feature.

### CODE REVIEW

**Review by Blackross22 — APPROVED** [review]

File: apiRoutes/authenticate.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    //Username not exist
+    if(findUser==null) return res.sendStatus(404)
+
+    let userData = findUser.dataValues
+    let userID = findUser.user_ID
+    let userPassword = userData.password
+
+    //Check password
+    if(userPassword == password){
+        let token = jwt.sign({ user_ID: userID}, "SECRET")
```
Comment (Icyscools) [inline apiRoutes/authenticate.js:29]: Please make `.env` ([enviroment file](https://www.npmjs.com/package/dotenv)) to set `SECRET` key.
to make sure that we don't have to manual change a `SECRET` key later.

File: authenticator.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+const jwt = require("jsonwebtoken")
+
+module.exports = function (req, res, next){
+
+    let authHeader = req.headers.authorization
+    let token = authHeader && authHeader.split(" ")[1]
+    //Empty token
+    if(token==null) return res.sendStatus(401)
+
+    jwt.verify(token, "SECRET", async (err, user)=>{
```
Comment (Icyscools) [inline authenticator.js:10]: Please make `.env` ([enviroment file](https://www.npmjs.com/package/dotenv)) to set `SECRET` key.
to make sure that we don't have to manual change a `SECRET` key later.

File: authenticator.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        //invalid tokens
+        if(err){
+            console.log(err)
+            return res.sendStatus(400)
+        }
+        req.body.user_ID = user.user_ID
+        console.log("\x1b[33m", "UserID: " + user.user_ID + "\x1b[37m")
```
Comment (Icyscools) [inline authenticator.js:17]: Remove console.log

**Review by Icyscools — COMMENTED** [review]
We create `.env` file to separate API environment between development and production.
In this case, we may change a `SECRET` key later after launch to production, 
so we'll need to change `SECRET` key for one time. Not all variable that use this key in the code.

**Review by Icyscools — APPROVED** [review]

**Pull request merged**